### ISSUE TITLE
Custom Player for Google IMA Ads

### ISSUE THREAD

**Issue by WINOFFRG** [shaka-project/shaka-player#4662]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
Yes

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
v4.2.3

**Please ask your question**
Hi! How do I create a custom Ad Player without using IMA ADs SDK? I see that it creates an Iframe source and then manually creates an overlay with video element interacting with our Shaka's AdManager. I plan to create a similar Ad Player like this which loads VAST Ads. However, I dont understand is there some parser script that we load or if provided externally who provides us the parsed VSAT Data and itself triggers the tracking events? I expect my custom AdPlayer to always load video files and present a static layout as shown below. 

Kindly help with this, In case there are any examples or references please drop that will be really helpful.

![image](https://user-images.githubusercontent.com/61311751/200534567-1bc0447e-8794-4b42-8cb8-0e30a1daec62.png)

**Comment by joeyparrish** [comment]
If you want to handle client-side ads yourself, or integrate with another ads SDK, you can do that on top of Shaka Player.  Here are some tips:

1. When it's time for an ad, you can pause the main video and overlay another video element for the ad.
2. If you can't have multiple video elements for some reason:
    1. capture the main video's `currentTime` and the player's `getAssetUri()`
    2. call `player.detach()` to ask it to detach from the video element
    3. load ad content in the video element
    4. wait for the ad to complete
    5. `await player.attach(videoElement)` to reattach to the video element
    6. `await player.load(assetUri, previousVideoTime)` to resume playback at the previous position

Does this help?

**Comment by WINOFFRG** [comment]
Thanks for your response. However my current plan is to create two different video elements and toggle between them. I definitely understand the plan you shared above, Thanks again. Can you help in letting me know, 

1. How do I parse VAST XML file recieved without using the IMA Ads SDK Script or any other (since they are generally iFrame loaded, which I definitely don't prefer to use). Is there any open source library sort of thing available out there, which can do this for us?
2. Commonly who is handling the ad events/trackers (ad-complete)? Should we do that ourself or some XYZ Script loaded externally should do this?

It would be great if you could please answer these as well.

**Comment by WINOFFRG** [comment]
Hi @joeyparrish Is there any update on this? Humbly waiting for your response.

**Comment by WINOFFRG** [comment]
Understood, Thanks a lot. Will be happy to drop updates for reference for other in case I find one.
